projects/debouncer/src/debouncer.c

```diff
@@ -0,0 +1,58 @@
+#include "debouncer.h"
+
+#include "log.h"
+#include "soft_timer.h"
+#include "stm32f0xx_interrupt.h"
+
+#define INTERRUPT_MASKING_DURATION_MS 50
+
+// This is the callback for the soft timer. Gets the state and compares with old state.
+// If states are equal, it means that we had a button input. So it runs the given callback.
+static void prv_debouncer_timer_callback(SoftTimerID timer_id, void *context) {
+  DebouncerInfo *debouncer_info = context;
+
+  GPIOState current_state;
+  gpio_get_state(&debouncer_info->address, &current_state);
+
+  if (current_state == debouncer_info->state) {
+    debouncer_info->callback(&debouncer_info->address, debouncer_info->context);
+  }
+
+  // stm32f0xx_interrupt_exti_mask_set(debouncer_info->address.pin, false);
+  gpio_it_mask_interrupt(&debouncer_info->address, false);
+}
+
+// Saves the state of gpio, masks the interrupt, and creates a soft timer
```

Comment:
Comments like these aren't that useful - they just tell you how the code works. The other problem is that if this function is ever modified and the comments aren't updated, it becomes very confusing.

I think a single top-level comment explaining how the pieces fit together would be more helpful.